### PR TITLE
Add Ubuntu 20.04 Support

### LINKED ISSUE #229: Doesn't work on ubuntu focal 20.04
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 6.16.0
- Ruby: 2.7.0
- Distribution: Ubuntu 20.04
- Module version: 5.1.0

## How to reproduce (e.g Puppet code you use)
```
class { 'snmp':
  agentaddress => [ 'udp:161', 'udp6:161' ],
}
```

## What are you seeing
Errors, see Output log below

## What behaviour did you expect instead
Deploys successfully without errors

## Output log
```
Error: Could not find user snmp
Error: /Stage[main]/Snmp/File[var-net-snmp]/owner: change from 'Debian-snmp' to 'snmp' failed: Could not find user snmp
Error: Could not find group snmp
Error: /Stage[main]/Snmp/File[var-net-snmp]/group: change from 'Debian-snmp' to 'snmp' failed: Could not find group snmp
Notice: /Stage[main]/Nrpe/Package[nagios-plugins]/ensure: created (corrective)
Notice: /Stage[main]/Snmp/Service[snmptrapd]: Dependency File[var-net-snmp] has failures: true
Warning: /Stage[main]/Snmp/Service[snmptrapd]: Skipping because of failed dependencies
Warning: /Stage[main]/Snmp/Service[snmpd]: Skipping because of failed dependencies
```

## Any additional information you'd like to impart
Looks like there a pull request already on https://github.com/voxpupuli/puppet-snmp/pull/226. Would be nice to have this verified and merged. I can see 3 tests failing there.

### PR DESCRIPTION
**Pull request by peelman**
#### Pull Request (PR) description
Invert snmp group and owner logic; support > 18.04 by default and exempt out Debian 7 & 8 and Ubuntu 14 and 16.  Comments or requests for amendments welcome.

#### This Pull Request (PR) fixes the following issues
Fixes #229

### CODE REVIEW

**Comment by peelman** [comment]
@bastelfreak rebase completed.  Sorry for the delay.

**Comment by kenyon** [comment]
I fixed that rebase. Make sure you update your fork's master branch before rebasing.

File: metadata.json [hunk 1]
```diff
@@ -62,7 +62,8 @@
       "operatingsystem": "Ubuntu",
       "operatingsystemrelease": [
         "16.04",
-        "18.04"
+        "18.04",
+        "20.04"
```
Comment (bastelfreak) [inline metadata.json:66]: can you also add ubuntu 20 to the .sync.yml and ubuntu 20 puppet 6 acceptance tests to the .travis.yml?

File: data/os/Debian/14.04.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (bastelfreak) [inline data/os/Debian/14.04.yaml:1]: we do not support 14.04 anymore and it's EOL, can you please remove this file?

File: data/os/Debian/7.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (bastelfreak) [inline data/os/Debian/7.yaml:1]: we don't support debian 7 anymore (and 8 will be dropped in the next days if I finde some free time), please don't add the yaml file for debian 7.

**Comment by aaroneg** [comment]
Hello! It would be nice to have this merged - is there anything I can do to help?

**Comment by kenyon** [comment]
Not sure why the Debian tests are failing, but Puppet 5 on Ubuntu 20.04 fails because those packages don't exist. Is there a way to exclude those tests somehow?

**Comment by kenyon** [comment]
Closed via #247.

**Pull request closed without merging**